botocore/httpsession.py

```diff
@@ -318,6 +333,11 @@ def _get_proxy_manager(self, proxy_url):
             proxy_manager_kwargs = self._get_pool_manager_kwargs(
                 proxy_headers=proxy_headers)
             proxy_manager_kwargs.update(**self._proxies_kwargs)
+            proxy_ssl_context = self._setup_proxy_ssl_context(self._proxy_config.settings, proxy_url)
+            if proxy_ssl_context:
```

Comment:
Before my change, the `proxies_kwargs` added `proxy_ssl_context` only when it was not `None`. This was also the problem behind tests failing in my last commit. `proxy_ssl_context` should not be in pool manager kwargs if it's None.

Reply:
@nateprewitt let me know if it's good now.

Reply:
Thanks @vabhasin, I've added a couple minor updates but I think we're set now. I've added @joguSD to do a final sign off since I've added changes. Then we'll look at getting the change merged.